runtime/lua/vim/health/health.lua

```diff
@@ -493,7 +493,153 @@ local function check_external_tools()
   end
 end
 
+local function detect_terminal()
+  local e = vim.env
+  if e.TERM_PROGRAM then
+    local v = e.TERM_PROGRAM_VERSION --- @type string?
+    return e.TERM_PROGRAM_VERSION and (e.TERM_PROGRAM .. ' ' .. v) or e.TERM_PROGRAM
+  end
+
+  local map = {
+    KITTY_WINDOW_ID = 'kitty',
+    ALACRITTY_SOCKET = 'alacritty',
+    ALACRITTY_LOG = 'alacritty',
+    WEZTERM_EXECUTABLE = 'wezterm',
+    KONSOLE_VERSION = function()
+      return 'konsole ' .. e.KONSOLE_VERSION
+    end,
+    VTE_VERSION = function()
+      return 'vte ' .. e.VTE_VERSION
+    end,
+  }
+
+  for key, val in pairs(map) do
+    local env = e[key] --- @type string?
+    if env then
+      return type(val) == 'function' and val() or val
+    end
+  end
+
+  return 'unknown'
+end
+
+local function check_sysinfo()
+  vim.health.start('System Info')
+
+  -- Use :version because vim.version().build returns "Homebrew" for brew installs.
+  local version_output = vim.api.nvim_exec2('version', { output = true }).output
+  local nvim_version = version_output:match('NVIM v[^\n]+') or 'unknown'
+  local commit_hash = nvim_version:match('%+g?(%x+)')
+  local version_for_report = nvim_version
+  if commit_hash then
+    version_for_report = nvim_version:gsub('%+g' .. commit_hash, ' neovim/neovim@' .. commit_hash)
+    local has_git = vim.fn.executable('git') == 0
+    local has_curl = vim.fn.executable('curl') == 1
+    local cmd = has_git and { 'git', 'ls-remote', 'https://github.com/neovim/neovim', 'nightly' }
+      or has_curl and {
+        'curl',
+        '-s',
+        'https://api.github.com/repos/neovim/neovim/commits/nightly',
+        '-H',
+        'Accept: application/vnd.github.sha',
+      }
+      or nil
+
+    if cmd then
+      local result = vim.system(cmd, { text = true }):wait()
+      if result.code == 0 then
+        local sha = has_git and result.stdout:match('^(%x+)') or result.stdout
+        local latest_commit = sha and sha:sub(1, 10)
+        if latest_commit and commit_hash ~= latest_commit then
+          vim.health.warn(
+            string.format('Build is outdated. Local: %s, Latest: %s', commit_hash, latest_commit)
```

Comment:
Er... why does this assume that `commit_hash ~= latest_commit` means "outdated" ? :)

Reply:
build hash does not match the nightly hash, does that indicate the build is outdated?

Reply:
Yes, but the message will almost always show "outdated", so people will stop noticing it. It will get ignored.

Counting commits would give a sense of how outdated they are.

Reply:
emm not exactly .. run neovim which i build yesterday it show me outdated 
<img width="1434" height="560" alt="image" src="https://github.com/user-attachments/assets/7be0681b-1322-4f3e-a5aa-0f9246678b3d" />

build with latest commit  there is no warning..

<img width="1204" height="538" alt="image" src="https://github.com/user-attachments/assets/26241ada-93d4-4fd8-8e09-bb0418f3315b" />

Reply:
I’ve thought about this before. I noticed that most PRs are merged during the night for people in certain time zones, so they rarely encounter merges during their daytime, which means the “outdated” warning doesn’t show up very often. There may not be a good solution for this. I just wanted a way to remind them to try building from the latest commit first and then verify whether the issue still exists, so  can avoid wasting the maintainers’ time 😮‍💨

Reply:
Check the release might be easier.

<img width="1528" height="1512" alt="image" src="https://github.com/user-attachments/assets/3a1777c4-a83c-4ea6-85d9-4d427c5fce36" />